samcli/lib/init/exceptions.py

```diff
@@ -16,5 +16,9 @@ class GenerateProjectFailedError(InitErrorException):
     fmt = "An error occurred while generating this project {project}: {provider_error}"
 
 
+class InvalidLocationError(InitErrorException):
+    fmt = "The template location specified is not valid: {template}"
```

Comment:
nit: since the class name has invalid, we could use the same in the message too.

```
The template location specified is invalid
```